packages/app/src/cli/models/extensions/specifications/flow_action.ts

```diff
@@ -1,35 +1,67 @@
 import {BaseSchema} from '../schemas.js'
 import {createExtensionSpecification} from '../specification.js'
+import {
+  validateNonCommerceObjectShape,
+  startsWithHttps,
+  validateCustomConfigurationPageConfig,
+} from '../../../services/Flow/validation.js'
+import {serializeFields} from '../../../services/Flow/serializeFields.js'
 import {joinPath} from '@shopify/cli-kit/node/path'
 import {glob, readFile} from '@shopify/cli-kit/node/fs'
-
 import {zod} from '@shopify/cli-kit/node/schema'
 
 const FlowActionExtensionSchema = BaseSchema.extend({
   name: zod.string(),
+  description: zod.string().optional(),
   type: zod.literal('flow_action'),
-  task: zod.object({
-    title: zod.string(),
-    description: zod.string(),
-    url: zod.string(),
-    validation_url: zod.string().optional(),
-    custom_configuration_page_url: zod.string().optional(),
-    custom_configuration_page_preview_url: zod.string().optional(),
-    schema: zod.string().optional(),
-    return_type_ref: zod.string().optional(),
-    fields: zod
-      .array(
-        zod.object({
-          id: zod.string(),
-          name: zod.string(),
-          label: zod.string(),
-          description: zod.string().optional(),
-          required: zod.boolean(),
-          ui_type: zod.string(),
-        }),
-      )
-      .optional(),
-  }),
+  extensions: zod
+    .array(
+      zod.object({
+        runtime_url: zod.string().url().refine(startsWithHttps),
+        validation_url: zod.string().url().refine(startsWithHttps).optional(),
+        config_page_url: zod.string().url().refine(startsWithHttps).optional(),
+        config_page_preview_url: zod.string().url().refine(startsWithHttps).optional(),
+        schema: zod.string().optional(),
+        return_type_ref: zod.string().optional(),
+      }),
+    )
+    .min(1),
+  settings: zod
+    .object({
+      fields: zod
```

Comment:
For both triggers and actions it looks like the `fields` array is optional. Tested in the Partner's Dashboard and through the CLI.